src/main/java/toughasnails/handler/thirst/FillBottleHandler.java

```diff
@@ -58,7 +58,7 @@ public void onRightClickCauldron(PlayerInteractEvent.RightClickBlock event)
         EntityPlayer player = event.getEntityPlayer();
         IBlockState state = world.getBlockState(event.getPos());
         
-        if (state.getBlock() instanceof BlockCauldron && SyncedConfig.getBooleanValue(GameplayOption.ENABLE_THIRST))
+        if (player.getHeldItem(event.getHand()).getItem() == Items.GLASS_BOTTLE && state.getBlock() instanceof BlockCauldron && SyncedConfig.getBooleanValue(GameplayOption.ENABLE_THIRST))
         {
```

Comment:
I am not sure if it was a bug with my code or not, but using a canteen on a cauldron would cause a watter bottle (not sure what kind, think it was filtered) to replace the canteen, this fixed it